### DISCUSSION TITLE
Any way to open the most closely matching file directly?

### DISCUSSION THREAD

**Discussion by niamleeson** [ibhagwan/fzf-lua#1156] [General]
Hello thank you for making this awesome plugin!

I'm trying to improve my workflow a little bit better. In my workflow, I work with 4 files:

```
components/**/my-component.js
components/**/my-component.hbs
tests/unit/**/my-component-test.js
tests/integration/**/my-component-test.js
```

If I'm on the component js file, I'd like to quickly navigate to other related files. But the files could be under different nested directories. So I'd like to use the current file name and some additional keywords to search for the related files and open the most closely matching one.

Right now, the commands open the file explorer. Is there a way to immediately open the most closely matching file?

Thank you!

**Comment by ibhagwan** [comment]
Fzf has an option to open the selected match if only one match is available, from `man fzf`:
```man
       -1, --select-1
              If there is only one match for the initial query (--query), do
              not start interactive finder and automatically select the only
              match
```

If you pre-supply the input text (query) and there will be only one match, fzf-lua will open the first selected match, for example:

> Using the vimL command:
```vim
:FzfLua files fzf_opts.--select-1=true query=compjs
```

> Using lua
```lua
:lua require("fzf-lua").files({fzf_opts={["--select-1"]=true},query="copmjs"})
```

The only limitation is that there must be **one match only**, if your `query` matches more than one file the interface will not be closed automatically.

**Comment by ibhagwan** [comment]
Another thing that might help your situation, in the recent few commits I added an option to format the results in a vscode-like filename first format, since fzf only cares about the string order (it is not aware of paths) this might help your match results (you can query the filename first), try it out with:

```vim
:FzfLua files formatter=path.filename_first
```

The result format will place the file part first and the folder later (grey color):
![image](https://github.com/ibhagwan/fzf-lua/assets/59988195/6a86a917-fab3-4d7e-87e1-9861b5e0607c)

**Reply by niamleeson** [reply]
That's brilliant! Thank you!
